jbmc/src/java_bytecode/assignments_from_json.cpp

```diff
@@ -257,25 +255,24 @@ static jsont get_untyped_string(const jsont &json)
 static optionalt<java_class_typet> given_runtime_type(
   const jsont &json,
   const optionalt<std::string> &type_from_array,
-  symbol_table_baset &symbol_table)
-{
-  std::string runtime_type;
-  if(!has_type(json) && !type_from_array)
-    return {};
-  if(has_type(json))
-    runtime_type = "java::" + json["@type"].value;
-  else
-  {
-    PRECONDITION(
+  const symbol_table_baset &symbol_table)
+{
+  const auto runtime_type = [&]() -> optionalt<std::string> {
+    if(const auto type_from_json = get_type(json))
+      return "java::" + *type_from_json;
+    if(!type_from_array)
+      return {};
```

Comment:
:pick: Why not pull this test out of the lambda? Obviously `get_untyped` should return `{}` whenever `type_from_array` is empty.

Reply:
Done. The `if(!type_from_json && !type_from_array)` check is done before the lambda now, and the lambda returns a std::string.